models/app.model.js

```diff
@@ -0,0 +1,11 @@
+const db = require("../db/connection");
+
+exports.selectTopics = () => {
+    return db.query("SELECT * FROM topics;")
+    .then(( { rows }) => {
+        if (rows.length === 0) {
+            return Promise.reject({ status: 404, msg: "No topics found"});
+        }
```

Comment:
I really like that this was considered, but should we be throwing an error if no topics exist?

In the case of our test data we should always expect there to be some topics, and if this is not the case I believe it is better for the 500 error to be thrown as something unexpected has happened, instead of a 404.

We should also always be trying to test for any code that we write as well and this is something that cant really be tested for as we would have to remove all the test data to do that as well.